include/sys/multi_heap.h

```diff
@@ -51,10 +51,16 @@ struct sys_multi_heap;
 typedef void *(*sys_multi_heap_fn_t)(struct sys_multi_heap *mheap, void *cfg,
 				     size_t align, size_t size);
 
+
+struct sys_heap_ext {
+	struct sys_heap *sysheap;
+	void *metadata;
+};
```

Comment:
Calling this "_ext" is a little confusing.  Usually that implies that it's an "extension" to the previous record, but this is a separate data structure that associates a heap with a user-supplied pointer.

Also, it's a 100% multiheap data structure, it's not part of sys_heap.  So I'd suggest naming more like:

```c
struct sys_multi_heap_rec {
    struct sys_heap *heap;
    void *user_data;
};
```